flask_app/static/js/friends-script.js

```diff
@@ -0,0 +1,48 @@
+let count = 0;
+let searchText = document.getElementById('friend-search-input').value;
+const user_id = document.getElementById('user-id-from-session').value;
+
+function handleTextInput(event){
+    searchText = event.target.value;
+    count++;
+    setTimeout(() => {
+        count--;
+        if( count > 0 ){
+            return;
+        }
+        friendsSearch(searchText, user_id);
+    }, 1000)
+}
+
+function searchButtonClick(event){
+    event.preventDefault();
+    friendsSearch(searchText, user_id);
+}
+
+function friendsSearch(searchText, user_id){
+    searchText = searchText.trim();
+    let element = document.getElementById('friend-search-results');
+    if( searchText.length < 1 ){
+        element.innerHTML = "";
+        return;
+    }
+    let innerHTML = ""
+    fetch('http://localhost:5000/api/friends/'+user_id+"/"+searchText )
+        .then( response => response.json() )
+        .then( data => {
+            ( data.message.length > 0 ?
+                data.message.map( friend => (
+                    innerHTML += "<div class='d-flex justify-content-center align-items-center m-2'>"
+                    +"<h3 class='mx-2'>"+friend.full_name+"</h3>"
```

Comment:
Are you absolutely sure you are sanitizing `friend.full_name`?

Reply:
Is there a specific way this needs to be 'sanitized' when passing names to javascript using a rest API?

I haven't taken any steps to 'sanitize' this data.

The Api returns a list of up to 5 users based on the search:
```js
data.message = [
    {"id": 1, "full_name": "person1 name"},
    {"id": 2, "full_name": "person2 name"}
]
```

Reply:
Using the UI, can you set a `full_name` to 

```html
<script>alert('XSS')</script>
```

for example.

Reply:
so that the data looked like 

```js
data.message = [
    {"id": 1, "full_name": "<script>alert('XSS')</script>"}
]
```

and would that render in the ui as a string or as a script?

Reply:
I am not super familiar with flask. This may be happening for you. Interpolating user provided strings in html is always risky, so you need to make sure things are being escaped at some point.

Reply:
Okay, this makes sense now. I just updated the user validation with a simple regex to prevent injecting scripts/tags like you mentioned.